src/webview/js/js.js

```diff
@@ -679,6 +680,14 @@ documentBody.addEventListener('click', (e: MouseEvent) => {
     return;
   }
 
+  if (target.closest('.spoiler-header')) {
+    const spoiler = target.closest('.spoiler-block');
```

Comment:
Because we only want to toggle the spoiler if the header is pressed.

Reply:
Just to see what it was like, I just tried only using `.spoiler-block`, so that the entire area of the spoiler was tappable while expanded, not just the header. I moved this code toward the bottom so things like links would continue to be handled.

It's different from the web app, which only has the header clickable, but first I thought it might be slightly more convenient to have a larger tappable area on mobile.

But then when I tried a [longer spoiler](https://chat.zulip.org/#narrow/stream/7-test-here/topic/spoiler/near/1058916), it occurred to me that spoilers might take up multiple screenfuls, and that the header won't always be in view—meaning one might sometimes forget that they're looking at a spoiler, and might in that case be frustrated or confused to see it close when they tap on it.

Reply:
> So I guess I'm slightly in favor of keeping only the header tappable, and that means looking for these two different elements—the header, to handle taps, and the whole block, to do the actual expanding and collapsing.

Hmm, I see.

That's a fine behavior choice. The place I'd want that distinction to live is inside the spoiler-specific code, in the implementation of `toggleSpoiler`. That function already has to do some navigating of the DOM in any case.

That way this `click` handler can remain focused on identifying if something clickable was clicked. So it'd end up looking only for `.spoiler-header`.